FlowCal/excel_ui.py

```diff
@@ -357,57 +360,78 @@ def process_beads_table(beads_table,
             beads_sample_gated = FlowCal.gate.start_end(beads_sample,
                                                         num_start=250,
                                                         num_end=100)
-            # Remove saturating events in forward/side scatter. The value of a
-            # saturating event is taken automatically from
-            # `beads_sample_gated.range`.
-            beads_sample_gated = FlowCal.gate.high_low(beads_sample_gated,
-                                                       channels=sc_channels)
+            # Remove saturating events in forward/side scatter, if the FCS data
+            # type is integer. The value of a saturating event is taken
+            # automatically from `beads_sample_gated.range`.
+            if beads_sample_gated.data_type == 'I':
+                beads_sample_gated = FlowCal.gate.high_low(
+                    beads_sample_gated,
+                    channels=sc_channels)
             # Density gating
+            # Axis scaling will be logarithmic if data was acquired with a log
+            # amplifier, and logicle otherwise.
+            sc_amp_type = beads_sample_gated.amplification_type(sc_channels)
             beads_sample_gated, __, gate_contour = FlowCal.gate.density2d(
                 data=beads_sample_gated,
                 channels=sc_channels,
                 gate_fraction=beads_row['Gate Fraction'],
-                xscale='log'
-                       if beads_sample_gated.amplification_type(
-                           sc_channels[0])[0]
-                       else 'linear',
-                yscale='log'
-                       if beads_sample_gated.amplification_type(
-                           sc_channels[1])[0]
-                       else 'linear',
+                xscale='log' if sc_amp_type[0][0] else 'logicle',
+                yscale='log' if sc_amp_type[1][0] else 'logicle',
```

Comment:
same as mentioned elsewhere: I think amplifier scale and plotting (and gating) scale are orthogonal, so I would still go with logicle gating, even with a log amplifier.